src/server/game/Entities/Unit/Unit.h

```diff
@@ -1866,8 +1939,10 @@ class TC_GAME_API Unit : public WorldObject
         void _ExitVehicle(Position const* exitPosition = nullptr);
         void _EnterVehicle(Vehicle* vehicle, int8 seatId, AuraApplication const* aurApp = nullptr);
 
-        void BuildMovementPacket(ByteBuffer* data) const;
-        static void BuildMovementPacket(Position const& pos, Position const& transportPos, MovementInfo const& movementInfo, ByteBuffer* data);
+        MovementInfo GetMovementInfo() const;
+        // should only be used by packet handlers to validate and apply incoming MovementInfos from clients. Do not use internally to modify m_movementInfo
+        void UpdateMovementInfo(MovementInfo movementInfo);
+        void ValidateNewMovementInfo(MovementInfo* mi);
```

Comment:
Similar function already exists in master branch under name "ValidateMovementInfo", please unify naming